src/chapter17/JavaFXEventDemo.java

```diff
@@ -0,0 +1,62 @@
+package chapter17;
+
+import javafx.application.Application;
+import javafx.stage.Stage;
+
+
+import javafx.application.*; 
+import javafx.scene.*; 
+import javafx.scene.layout.*; 
+import javafx.scene.control.*; 
+import javafx.geometry.*; 
+ 
+public class JavaFXEventDemo extends Application { 
+ 
+  Label response; 
+ 
+  public static void main(String[] args) { 
+ 
+    // Start the JavaFX application by calling launch(). 
+    launch(args);   
+  } 
+ 
+  // Override the start() method. 
+  public void start(Stage myStage) { 
+ 
+    // Give the stage a title. 
+    myStage.setTitle("Use JavaFX Buttons and Events."); 
+ 
+    // Use a FlowPane for the root node. In this case, 
+    // vertical and horizontal gaps of 10. 
+    FlowPane rootNode = new FlowPane(10, 10); 
+ 
+    // Center the controls in the scene. 
+    rootNode.setAlignment(Pos.CENTER); 
+ 
+    // Create a scene. 
+    Scene myScene = new Scene(rootNode, 300, 100); 
+ 
+    // Set the scene on the stage. 
+    myStage.setScene(myScene); 
+ 
+    // Create a label. 
+    response = new Label("Push a Button"); 
+ 
+    // Create two push buttons. 
+    Button btnUp = new Button("Run"); 
+    Button btnDown = new Button("Exit"); 
+ 
+    // Handle the action events for the Run button. 
+    btnUp.setOnAction( (ae) -> response.setText( "You pressed Run" ) ); 
+ 
+    // Handle the action events for the Exit button. 
+    btnDown.setOnAction( (ae) -> Platform.exit() ); 
```

Comment:
For single arg lambdas, you can leave out the parentheses, if you like (or if you're in some kind of parenthesis famine): :cold_sweat: 
```java
ae -> Platform.exit()
```